File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/domain/interactors/types/UseCaseSingle.java
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.interactors.types;

import io.reactivex.Single;

public interface UseCaseSingle<T, Params> {

    Single<T> runUseCase(Params... params);
}

File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/domain/interactors/types/UseCaseSingle.kt
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.interactors.types

import io.reactivex.Single

interface UseCaseSingle<T, Params> {

    fun runUseCase(vararg params: Params): Single<T>
}
